torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -456,12 +456,13 @@ def _cast_fp_inputs_to_dtype(
     def cast_fn(x: torch.Tensor) -> torch.Tensor:
         if not torch.is_floating_point(x) or x.dtype == dtype:
             return x
-        y = x.to(dtype)
-        # Explicitly copy over `requires_grad` since this runs inside
-        # `torch.no_grad()`
```

Comment:
I think this comment about `_cast_fp_inputs_to_dtype` running in `no_grad` is outdated given my check just now.